api/v1beta1/imagerepository_types.go

```diff
@@ -0,0 +1,147 @@
+	// +optional
+	// +kubebuilder:validation:Pattern="^[a-z0-9][.a-z0-9_-]*(/[a-z0-9][.a-z0-9_-]*)*$"
+	Name string `json:"name,omitempty"`
+
+	// Visibility defines whether the image is publicly visible.
+	// Allowed values are public and private.
+	// "public" is the default.
+	// +optional
+	Visibility ImageVisibility `json:"visibility,omitempty"`
+}
+
+// +kubebuilder:validation:Enum=public;private
+type ImageVisibility string
+
+const (
+	ImageVisibilityPublic  ImageVisibility = "public"
+	ImageVisibilityPrivate ImageVisibility = "private"
+)
+
+type ImageCredentials struct {
+	// RegenerateToken defines a request to refresh image accessing credentials.
+	// Refreshes both, push and pull tokens.
+	// The field gets cleared after the refresh.
+	RegenerateToken *bool `json:"regenerate-token,omitempty"`
+}
+
+// ImageRepositoryStatus defines the observed state of ImageRepository
+type ImageRepositoryStatus struct {
+	// State shows if image repository could be used.
+	// "ready" means repository was created and usable,
+	// "failed" means that the image repository creation request failed.
+	State ImageRepositoryState `json:"state,omitempty"`
+
+	// Message shows error information for the request.
+	// It could contain non critical error, like failed to change image visibility,
+	// while the state is ready and image resitory could be used.
+	Message string `json:"message,omitempty"`
+
+	// Image describes actual state of the image repository.
+	Image ImageStatus `json:"image,omitempty"`
+
+	// Credentials contain information related to image repository credentials.
+	Credentials CredentialsStatus `json:"credentials,omitempty"`
+}
+
+type ImageRepositoryState string
+
+const (
+	ImageRepositoryStateReady  ImageRepositoryState = "ready"
+	ImageRepositoryStateFailed ImageRepositoryState = "failed"
+)
+
+// ImageStatus shows actual generated image repository parameters.
+type ImageStatus struct {
+	// URL is the full image repository url to push into / pull from.
+	URL string `json:"url,omitempty"`
+
+	// Visibility shows actual generated image repository visibility.
+	// +kubebuilder:validation:Enum=public;private
+	Visibility ImageVisibility `json:"visibility,omitempty"`
+}
+
+// CredentialsStatus shows information about generated image repository credentials.
+type CredentialsStatus struct {
+	// GenerationTime shows timestamp when the current credentials were generated.
+	GenerationTimestamp *metav1.Time `json:"generationTimestamp,omitempty"`
+
+	// PushSecretName holds name of the dockerconfig secret with credentials to push (and pull) into the generated repository.
+	PushSecretName string `json:"push-secret,omitempty"`
+
+	// PullSecretName is present only if ImageRepository has labels that connect it to Application and Component.
+	// Holds name of the dockerconfig secret with credentials to pull only from the generated repository.
+	// The secret is not present in the same namespace as ImageRepository, but created in
```

Comment:
Looks like the javadoc just trails off here?

Reply:
Sort of